libminifi/include/core/TypedValues.h

```diff
@@ -149,10 +144,39 @@ class DataSizeValue : public TransformableValue, public state::response::UInt64V
   }
 };
 
-}  // namespace core
-}  // namespace minifi
-}  // namespace nifi
-}  // namespace apache
-}  // namespace org
+class DataTransferSpeedValue : public TransformableValue, public state::response::UInt64Value {
+ public:
+  MINIFIAPI static const std::type_index type_id;
+
+  explicit DataTransferSpeedValue(const std::string &transfer_speed_string)
+      : state::response::UInt64Value(0) {
+    DataSizeValue::StringToInt<uint64_t>(removePerSecSuffix(transfer_speed_string), value);
+    string_value = transfer_speed_string;
+  }
+
+  explicit DataTransferSpeedValue(uint64_t value)
+      : state::response::UInt64Value(value) {
+  }
+
+  DataTransferSpeedValue()
+      : state::response::UInt64Value(0) {
+  }
+
+  static std::string removePerSecSuffix(const std::string &input) {
+    auto lower_case_input = utils::StringUtils::toLower(input);
+    if (lower_case_input.ends_with("/sec")) {
+      return input.substr(0, input.size() - 4);
+    }
```

Comment:
I'd prefer to keep the set of accepted formats smaller, and only accept "/s", not "/sec" or "ps". It's easy to extend later, but it harder to break what worked before.

Reply:
Updated in f8833c3b01c5edf0ff60fff593b2f21ab91763cd